app/frontend/src/features/FlagButton.tsx

```diff
@@ -0,0 +1,195 @@
+import {
+  FormControl,
+  FormHelperText,
+  IconButton,
+  InputLabel,
+  Select,
+} from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "components/Dialog";
+import { FlagIcon } from "components/Icons";
+import Snackbar from "components/Snackbar";
+import TextField from "components/TextField";
+import {
+  CANCEL,
+  CONTENT_REPORT,
+  CONTENT_REPORT_DESCRIPTION_HELPER,
+  CONTENT_REPORT_DESCRIPTION_LABEL,
+  CONTENT_REPORT_EXPLAINER,
+  CONTENT_REPORT_REASON_HELPER,
+  CONTENT_REPORT_REASON_LABEL,
+  CONTENT_REPORT_REASONS,
+  CONTENT_REPORT_SUCCESS,
+  REASON_REQUIRED,
+  SUBMIT,
+} from "features/constants";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error as GrpcError } from "grpc-web";
+import { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+import { service } from "service";
+import { ReportInput } from "service/reporting";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  typeButton: {
+    display: "block",
+    margin: "0 auto",
+    "& + &": {
+      marginBlockStart: theme.spacing(2),
+    },
+  },
+  field: {
+    "& + &": {
+      marginBlockStart: theme.spacing(2),
+    },
+  },
+  formControl: {
+    "& .MuiOutlinedInput-root": {
+      borderRadius: theme.shape.borderRadius * 3,
+    },
+  },
+}));
+
+export interface ReportButtonProps {
+  contentRef: string;
+  authorUser: string | number;
+}
+
+export default function ReportButton({
```

Comment:
```suggestion
export default function FlagButton({
```
For consistency with the component name